integration-tests/bats/sql-branch.bats

```diff
@@ -162,3 +162,13 @@ SQL
     run dolt sql -q "CALL DOLT_BRANCH('-D', 'branch_with_unpushed_commit');"
     [ $status -eq 0 ]
 }
+
+@test "sql-branch: CALL DOLT_BRANCH -d error cases" {
```

Comment:
Do we need an equivalent test for the sql command? What should the behavior be there?

Reply:
Good catch. I added a new BATS test for the `dolt branch -D` to assert that you cannot delete the checked out branch. _(I know you said for the sql command, but I generalized to make sure we have all the branch delete codepaths covered.)_

Now we've got the following cases covered:
- dolt sql shell ([sql-branch.bats](https://github.com/dolthub/dolt/blob/ae23057590ce6a27e615e2d51a6a7669a5541b9a/integration-tests/bats/sql-branch.bats#L166))
- dolt sql-server connection ([sql-server.bats](https://github.com/dolthub/dolt/blob/ae23057590ce6a27e615e2d51a6a7669a5541b9a/integration-tests/bats/sql-server.bats#L575))
- dolt CLI ([branch.bats](https://github.com/dolthub/dolt/blob/40965c6d7a7544354f6bf5f312cad491ea6661ca/integration-tests/bats/branch.bats#L142))